test/extensions/implementation/autograd.py

```diff
@@ -176,3 +177,42 @@ def diag_h_batch(self):
         factor = self.problem.get_reduction_factor(batch_loss, loss_list)
         params_batch_diag_h = list(zip(*batch_diag_h))
         return [torch.stack(param) * factor for param in params_batch_diag_h]
+
+    def ggn(self):
```

Comment:
Public method without docstring. So either 
- add docstring
- or `# noqa: D102` and docstring to parent class

Same applies to the other methods

Reply:
All public methods are documented in `test/extensions/implementation/base.py`. Adding type hints and `noqa`s. Adding the file to `fully_documented.txt`.